admin/server/auth/handlers.go

```diff
@@ -140,7 +167,16 @@ func (a *Authenticator) authStart(w http.ResponseWriter, r *http.Request, signup
 		return
 	}
 
-	// Redirect to auth provider
+	// Redirect to <canonical-domain>/auth/login (custom domain flow)
+	if a.admin.URLs.IsCustomDomain(r.Host) {
+		// Redirect to canonical domain with custom domain callback as redirect
+		customCallbackURL := a.admin.URLs.External() + "/auth/custom-domain-callback?state=" + state
```

Comment:
It should not do URL building using string concats, it should do it with a new func on `admin.URLs` and use the proper URL building utils.